### PR TITLE
feat(home-manager): add support for hyprlock

### PR DESCRIPTION
**Pull request by PerchunPak**
https://github.com/catppuccin/hyprlock

### CODE REVIEW

File: modules/home-manager/hyprlock/conf.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     blur_passes = 0
+     color = $base
+ }
+@@ -42,19 +37,6 @@ label {
+     valign = top
+ }
+ 
+-# USER AVATAR
+-
+-image {
```
Comment (PerchunPak) [inline modules/home-manager/hyprlock/conf.patch:27]: As far as I understand, there is no consistent way to get user's avatar, so I removed it to make it work from the box

File: modules/home-manager/hyprlock/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      # define accent color
+      (builtins.toFile "hyprlock-${cfg.accent}.conf" ''
+        $accent=''$${cfg.accent}
+        $accentAlpha=''$${cfg.accent}Alpha
+      '')
+
+      # source port with patches to make it work from the box
+      (pkgs.runCommand "hyprlock-catppuccin.conf" { } ''
+        cat ${sources.hyprlock}/hyprlock.conf > $out
+        patch -p1 $out ${./conf.patch}
```
Comment (PerchunPak) [inline modules/home-manager/hyprlock/default.nix:31]: BTW, do note that GNU patch needs trailing spaces to work.

**Comment by kcoderhtml** [comment]
oooh; thank you so much! i can't wait for this to be integrated!

**Comment by PerchunPak** [comment]
Closing this PR as the author of the theme doesn't support LGBTQ movement.

**Pull request closed without merging**